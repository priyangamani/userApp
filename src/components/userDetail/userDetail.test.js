import {
  render,
  cleanup,
  within,
} from "@testing-library/react";
import UserDetail from "./userDetail";

describe("<App />", () => {
  afterEach(() => {
    cleanup();
  });

  const props = {
    name: "username",
    email: "email",
    phone: "+65-12333",
  };


  test('should take a snapshot', () => {
    const { asFragment } = render(<UserDetail {...props}/>)
    expect(asFragment(<UserDetail />)).toMatchSnapshot()
   })

   test('render props', () => {
    const { getByTestId } = render(<UserDetail {...props}/>)
    const { getByText } = within(getByTestId('name'))
    expect(getByText(props.name)).toBeInTheDocument()
   })
});
