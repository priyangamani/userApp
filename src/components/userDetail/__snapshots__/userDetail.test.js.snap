// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="box-container"
  >
    <div
      class="box"
    >
      <b>
        Name: 
      </b>
      <a
        data-testid="name"
      >
        username
      </a>
      <br />
      <b>
        Email: 
      </b>
      <a
        data-testid="email"
        href="mailto: email"
      >
        email
      </a>
      <br />
      <b>
        Telephone: 
      </b>
      <a
        data-testid="phone"
        href="tel:+65-12333"
      >
         +65-12333 
      </a>
      <br />
    </div>
  </div>
</DocumentFragment>
`;
